plugins/generator-addon-1.21.x/addon-1.21.x/templates/item.json.ftl

```diff
@@ -9,17 +9,17 @@
     },
 
     "components": {
-      <#if data.glowCondition?? && data.glowCondition.getFixedValue()>"minecraft:foil": true,</#if>
+      "minecraft:foil": ${data.hasGlint},

```

Comment:
It would be better to omit component if it does not have glint like before, so component is not applied if not needed